### ISSUE TITLE
Bug in current develop with recurse

### ISSUE THREAD

**Issue by mathomp4** [GEOS-ESM/mepo#330]
I was trying to clone GEOS using mepo `develop` and it died with:
```
mom                    | (t) geos/5.1.0+1.2.0
fatal: invalid filter-spec 'blob:none--recurse-submodules'

Traceback (most recent call last):
  File "/Users/mathomp4/.local/bin/mepo", line 8, in <module>
    sys.exit(main())
             ~~~~^^
  File "/Users/mathomp4/.local/pipx/venvs/mepo/lib/python3.13/site-packages/mepo/__main__.py", line 15, in main
    cmd_module.run(args)
    ~~~~~~~~~~~~~~^^^^^^
  File "/Users/mathomp4/.local/pipx/venvs/mepo/lib/python3.13/site-packages/mepo/command/clone.py", line 41, in run
    clone_components(allcomps, arg_partial)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mathomp4/.local/pipx/venvs/mepo/lib/python3.13/site-packages/mepo/command/clone.py", line 107, in clone_components
    git.clone(version, recurse_submodules, partial)
    ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/mathomp4/.local/pipx/venvs/mepo/lib/python3.13/site-packages/mepo/git.py", line 60, in clone
    shellcmd.run(shlex.split(cmd))
    ~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^
  File "/Users/mathomp4/.local/pipx/venvs/mepo/lib/python3.13/site-packages/mepo/utilities/shellcmd.py", line 16, in run
    result.check_returncode()
    ~~~~~~~~~~~~~~~~~~~~~~~^^
  File "/Users/mathomp4/.homebrew/brew/Cellar/python@3.13/3.13.1/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 508, in check_returncode
    raise CalledProcessError(self.returncode, self.args, self.stdout,
                             self.stderr)
subprocess.CalledProcessError: Command '['git', 'clone', '--filter=blob:none--recurse-submodules', '--quiet', 'git@github.com:GEOS-ESM/MOM6.git', '/Users/mathomp4/Models/GEOSgcm-GCC14/GEOSgcm/src/Components/@GEOSgcm_GridComp/GEOSogcm_GridComp/@GEOS_OceanGridComp/MOM6_GEOSPlug/@mom6']' returned non-zero exit status 128.
```

From what it looks like, there is a space missing or something:
```
--filter=blob:none--recurse-submodules
```

**Comment by pchakraborty** [comment]
I'll try to have a test for it as well.

**Comment by pchakraborty** [comment]
Fixed in #331. Closing.
